src/node.cc

```diff
@@ -947,16 +947,46 @@ void* ArrayBufferAllocator::Allocate(size_t size) {
 }
 
 
-static bool IsDomainActive(const Environment* env) {
+static bool hasDomainErrorHandler(const Environment* env) {
+  HandleScope scope(env->isolate());
+
   if (!env->using_domains())
     return false;
 
-  Local<Array> domain_array = env->domain_array().As<Array>();
-  if (domain_array->Length() == 0)
+  Local<Array> domains_stack_array = env->domains_stack_array().As<Array>();
+  if (domains_stack_array->Length() == 0)
     return false;
 
-  Local<Value> domain_v = domain_array->Get(0);
-  return !domain_v->IsNull();
+  uint32_t domains_stack_length = domains_stack_array->Length();
+  for (int i = domains_stack_length - 1; i >= 0; --i) {
+    Local<Value> domain_v = domains_stack_array->Get(i);
+    if (domain_v->IsNull())
+      return false;
+
+    Local<Object> domain = domain_v->ToObject();
+    if (domain->IsNull())
+      return false;
+
+    Local<Value> domain_event_listeners_v = domain->Get(env->events_string());
+    if (!domain_event_listeners_v->IsObject())
+      return false;
+
+    Local<Object> domain_event_listeners_o =
+      domain_event_listeners_v->ToObject();
+
+    if (domain_event_listeners_o->IsNull())
+      return false;
+
+    Local<Value> domain_error_listeners_v =
+      domain_event_listeners_o->Get(env->error_string());
```

Comment:
style nit: these cases should be indented 4 spaces. the linter not catching this?